### PR TITLE
Feat/88  career card component

### LINKED ISSUE #88: [FEAT] External Dynamic Pages >Career Page > create the career card component
## Description
Develop a reusable career card component to display individual job listings on the Career Page. This component should present key job information in a clear, attractive, and consistent format.

## Acceptance Criteria

- The career card component accurately summarizes all required job information.
- The design is responsive and maintains visual consistency across different screen sizes.
- The component includes interactive elements that function correctly.

### Requirements	
The career card should include the following elements:
- Company location
- Tags for job type (e.g., "Remote") and department
- Brief job description or responsibilities
- Salary range
- "View details" button

### Tests

- Verify that the component renders correctly with all required information.
- Test the responsiveness of the card across various device sizes.
- Ensure the "View details" button functions as expected.
- Check that long text (e.g., job titles or descriptions) is handled appropriately.

###  Tasks

- Design the career card component based on the provided example and requirements.
- Implement the component structure using semantic HTML.
- Create props or data structure to populate the card with dynamic content.
- Add hover and focus states for interactive elements.
- Implement loading states or skeletons for when data is being fetched.

# LINKS

[`FIGMA LINK`](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1666-52483&m=dev).

# IMAGES

image

### PR DESCRIPTION
**Pull request by YeeshaDev**
# Description

**Closes #88**

# Changes proposed

## What were you told to do?

I was told to create the career card component for career page

## What did you do?

- I created the career card component using Shadcn UI library and tailwind
- I added the skeleton loading state so the card looks presentable before the datas are finall fetched from the server.
- I ensured and tested responsiveness acreoss all screens as asked.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

Macbook-Air-localhost
Galaxy-Tab-S7-localhost
iPhone-13-PRO-localhost
Galaxy-Tab-S7-localhost (1)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,12 +15,14 @@
...
   },
   "dependencies": {
+    "@radix-ui/react-slot": "^1.1.0",
     "@t3-oss/env-nextjs": "^0.10.1",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.1",
     "lucide-react": "^0.400.0",
     "next": "14.2.4",
     "next-auth": "5.0.0-beta.19",
+    "pnpm": "^9.5.0",
```
Comment (michaelhpet) [inline package.json]: Why are you installing pnpm in the project? pnpm should be on your system, not the project

**Review by michaelhpet — CHANGES_REQUESTED** [review]

**Comment by michaelhpet** [comment]
You have merge conflicts. Please sync your branch with the upstream

File: package.json (hunk 1 above)
Comment (YeeshaDev) [reply package.json]: fixed

File: src/components/common/CareerCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// pages/index.tsx
```
Comment (michaelhpet) [inline src/components/common/CareerCard/index.tsx]: no comments

File: src/components/common/CareerCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"use client";
+
+import React, { useEffect, useState } from "react";
+
+import CareerCard from "./CareerCard";
+
+//The View details button function
```
Comment (michaelhpet) [inline src/components/common/CareerCard/index.tsx]: ditto

File: src/components/common/CareerCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+const handleViewDetails = () => {
+  console.log("View career details!");
+};
+
+const CareerCardParent: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    // Simulate loading
```
Comment (michaelhpet) [inline src/components/common/CareerCard/index.tsx]: remove comments

File: src/test/Career/CareerCard.test.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+// components/CustomCard.test.tsx
```
Comment (michaelhpet) [inline src/test/Career/CareerCard.test.tsx]: remove comments

File: tailwind.config.ts [hunk 6]
```diff
@@ -20,12 +20,14 @@ const config = {
     extend: {
       colors: {
         border: "hsl(var(--border))",
+        "card-border": "#FAFAFA",
```
Comment (michaelhpet) [inline tailwind.config.ts]: use variables in hsl, see `globals.css`

File: tailwind.config.ts [hunk 7]
```diff
@@ -20,12 +20,14 @@ const config = {
...
+        cardText: "#0A0A0A",
```
Comment (michaelhpet) [inline tailwind.config.ts]: ditto, no rgb. use css variables

File: tailwind.config.ts [hunk 8]
```diff
@@ -20,12 +20,14 @@ const config = {
...
         input: "hsl(var(--input))",
         ring: "hsl(var(--ring))",
         background: "hsl(var(--background))",
         foreground: "hsl(var(--foreground))",
         primary: {
-          DEFAULT: "hsl(var(--primary))",
+          DEFAULT: "#F97316",
```
Comment (michaelhpet) [inline tailwind.config.ts]: ditto

**Review by michaelhpet — CHANGES_REQUESTED** [review]

File: tailwind.config.ts (hunk 6 above)
Comment (YeeshaDev) [reply tailwind.config.ts:23]: Sorry but I did not add the hsl styles that is how the dev branch is the only changes I made are the rgb (cardText and card-border)

File: tailwind.config.ts (hunk 7 above)
Comment (YeeshaDev) [reply tailwind.config.ts]: fixed

File: tailwind.config.ts (hunk 8 above)
Comment (YeeshaDev) [reply tailwind.config.ts]: fixed

File: tailwind.config.ts (hunk 6 above)
Comment (YeeshaDev) [reply tailwind.config.ts]: fixed

File: src/components/common/CareerCard/index.tsx (hunk 4 above)
Comment (YeeshaDev) [reply src/components/common/CareerCard/index.tsx]: fixed

File: src/test/Career/CareerCard.test.tsx (hunk 5 above)
Comment (YeeshaDev) [reply src/test/Career/CareerCard.test.tsx]: fixed

File: src/components/common/CareerCard/index.tsx (hunk 2 above)
Comment (YeeshaDev) [reply src/components/common/CareerCard/index.tsx]: fixed

**Pull request closed without merging**